tests/subsys/pm/device_runtime_api/src/main.c

```diff
@@ -190,6 +191,30 @@ static void test_api(void)
 
 	(void)pm_device_state_get(dev, &state);
 	zassert_equal(state, PM_DEVICE_STATE_ACTIVE, NULL);
+
+	/* Test state lock API */
+	pm_device_state_lock(dev);
+
+	(void)pm_device_state_get(dev, &state);
+	zassert_equal(state, PM_DEVICE_STATE_ACTIVE, NULL);
+
+	/* Put operation should failed due the state lock */
+	ret = pm_device_runtime_put(dev);
+	zassert_equal(ret, -EPERM, NULL);
+
+	locked = pm_device_state_is_locked(dev);
+	zassert_equal(locked, true, NULL);
```

Comment:
this check can be removed, it's not relevant for this test (this test is just checking runtime behaves correctly when state is locked, not the lock API)

Reply:
I don't agree here, if the runtime operation failed and the state is not locked, we have a problem and the test will catch it. Otherwise the put may fail (not because the state is locked) and we will accept.

Reply:
My point is: `device_runtime_api` tests the runtime API behavior. If the device_runtime_api behavior depends on `pm_device_state_is_locked`, then the runtime tests needs to be updated. However, if `pm_device_state_is_locked` behavior depends on the runtime API, then that should be part of a separate test that checks `pm_device_state*` behavior. According to final changeset, the runtime PM API is not affected by this new API, so runtime tests don't need to be updated. I know some PM tests mix responsibilities, but that should be gradually cleaned up, otherwise it's really difficult to understand the purpose of tests.

Reply:
yep, that is right. Was thinking in the initial behavior.